observer/observer.go

```diff
@@ -43,18 +43,21 @@ func (o *Observer) processBlock(events chan<- Event, block *blockatlas.Block) {
 	}
 
 	// Emit events
-	emitted := make(map[string]bool)
+	emittedUtxo := make(map[string]blockatlas.Direction)
 	platform := bitcoin.UtxoPlatform(o.Coin)
 	for _, sub := range subs {
+
 		txs := txMap[sub.Address].Txs()
 		for _, tx := range txs {
-			if _, ok := emitted[tx.ID]; ok {
-				continue
-			}
+
 			xpub, _ := o.Storage.GetXpubFromAddress(o.Coin, sub.Address)
-			if len(xpub) != 0 {
+			if len(xpub) > 0 {
+				xpubAddresses, err := o.Storage.GetAddressFromXpub(o.Coin, xpub)
+				if err != nil {
```

Comment:
can we move out some of the logic and add unit testing? to make sure direction handled properly

Reply:
This tests already exist, the problem was the parameters passed to the method were wrong

https://github.com/trustwallet/blockatlas/blob/ad803284bceff444bc4dab9752dcc892e304e466/platform/bitcoin/api_test.go#L197